src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -0,0 +1,280 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "boolean_funcs.h"
+#include "../func_desc.h"
+#include "../../util/arr.h"
+#include <assert.h>
+
+#define CONTAINS_NULL 2 // Macro used for efficiently evaluating 3-valued truth table
```

Comment:
Why `2`?

Reply:
Both arguments are either 1 or 0, so rightmost-bit only. 2 lets me play with the next bit to the left. We can use `1<<1` if you prefer?

I admit that the handling for 3-valued truth tables I've introduced here is a little annoyingly clever, but it avoids _a lot_ of conditional logic.